controllers/operator_controller.go

```diff
@@ -314,6 +314,16 @@ func verifyBDStatus(dep *rukpakv1alpha1.BundleDeployment) (metav1.ConditionStatu
 
 // mapBDStatusToReadyCondition returns the operator object's "TypeReady" condition based on the bundle deployment statuses.
 func mapBDStatusToReadyCondition(existingBD *rukpakv1alpha1.BundleDeployment, observedGeneration int64) metav1.Condition {
+	// if BundleDeployment status is stale, return an unknown condition.
+	if isBundleDepStale(existingBD) {
+		return metav1.Condition{
+			Type:               operatorsv1alpha1.TypeReady,
+			Status:             metav1.ConditionUnknown,
+			Reason:             operatorsv1alpha1.ReasonInstallationStatusUnknown,
+			Message:            fmt.Sprintf("waiting for bundleDeployment %q status to be updated. BundleDeployment conditions out of date.", existingBD.Name),
```

Comment:
Here, when both conditions are nil, I think we may now have a slightly misleading condition message: https://github.com/operator-framework/operator-controller/blob/cc2c7375262ce0a225ea9d00351ea1bcc498ced3/controllers/operator_controller.go#L297-L299

Before, we were assuming "those conditions not being present means it just hasn't reconciled yet"

Now, we know that the BD has been reconciled because we've made it past the staleness check unscathed. The status is up-to-date, but _still_ the expected conditions aren't present. I'd suggest just removing lines 297-299, and then if both conditions are `nil`, we'll end up at line 312, which seems more appropriate now I think. WDYT?